tests/test_model_evaluation.py

```diff
@@ -68,3 +69,22 @@ def test_write_nn_tc():
 
         true_outcomes = pd.read_csv(test_dir / "write_nn_tc.csv")
         pd.testing.assert_frame_equal(outcomes, true_outcomes)
+
+
+def test_train_discriminator():
+    with tempfile.TemporaryDirectory() as temp_dir:
+        output_file = Path(temp_dir) / "train_discriminator.csv"
+        outcomes = train_discriminator(
+            train_file=test_dir
+            / "snakemake_output/0/prior/inputs/train_LOTUS_truncated_SMILES_all.smi",
+            sample_file=test_dir
+            / "snakemake_output/0/prior/samples/LOTUS_truncated_SMILES_processed_freq-avg.csv",
+            output_file=output_file,
+            seed=0,
+        )
+
+        # Remove indexes of the result to match with the output file
+        outcomes = outcomes.reset_index(drop=True)
```

Comment:
This should happen inside the function before returning the `DataFrame` to the caller.

Reply:
Updated the code accordingly.